src/Autocomplete.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [noSuggestions, setNoSuggestions] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (inputValue.trim() === '') {
```

Comment:
According to the checklist, the filter function shouldn't be called if the user entered spaces only. However, in this case, if the input value is only spaces, you will set all people as suggestions. To avoid unnecessary filtering, you should also check if the input value is not composed of spaces only.